### DISCUSSION TITLE
Differences between srsRAN_P GNB vs srsRAN-4G GNB ( HARQ )

### DISCUSSION THREAD

**Discussion by dbsnooper** [srsran/srsRAN_Project#69] [General Help]
I am curious what major differences exist in the two GNB implementations other than ORAN/split related differences.

I just started swapping over from srsRAN[_4G] to srsRAN_Project and have unexpected issues as soon as the srsUE client attaches. From the command line srsUE does not proceed beyond "RRC Connected" when attaching to a seemingly identically configured *nodeB.

In both cases the gNodeB registers with the Open5Gs AMF and receives the log entries on that system.
The RAN and the Core systems are on separate machines, so I can get a pretty clean pcap (couldn't upload here). With srsenb, normal looking NGAP/NAS-5GS messages are exchanged. With gnb, NO MESSAGES are exchanged after the NGSetupResponse from Open5Gs. (ignoring heartbeats for both of those setups).

The only clear issue seems to involve HARQ, but unsure what steps to take in troubleshooting. Any help welcomed, thank you.

[outputs.txt](https://github.com/srsran/srsRAN_Project/files/11410406/outputs.txt)
[enb.log](https://github.com/srsran/srsRAN_Project/files/11410352/enb.log)
[gnb.log](https://github.com/srsran/srsRAN_Project/files/11410353/gnb.log)
[ue+enb.log](https://github.com/srsran/srsRAN_Project/files/11410354/ue%2Benb.log)
[ue+gnb.log](https://github.com/srsran/srsRAN_Project/files/11410355/ue%2Bgnb.log)

**Comment by pgawlowicz** [comment]
Hi,

You can follow this [tutorial](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html) that shows how to run srsUE with the new gNB.

Moreover, it should be possible to run the srsUE with the same config against the old and the new gNB. 

Could you share your configs?

Best,
Piotr

**Comment by dbsnooper** [comment]
Piotr,

Thank you for taking the time to help! (I forgot to attach the configs, but knew to do so. It's difficult getting old :p ) Attached here.

[n3.gnb.yaml.txt](https://github.com/srsran/srsRAN_Project/files/11431159/n3.gnb.yaml.txt)
[enb.conf.n3.txt](https://github.com/srsran/srsRAN_Project/files/11431158/enb.conf.n3.txt)
[rr.conf.n3.txt](https://github.com/srsran/srsRAN_Project/files/11431160/rr.conf.n3.txt)
[ue.conf.n3.txt](https://github.com/srsran/srsRAN_Project/files/11431161/ue.conf.n3.txt)

**Comment by pgawlowicz** [comment]
Hi, 

Please find attach a set of  [configs.zip](https://github.com/srsran/srsRAN_Project/files/11454400/configs.zip) that allow to connect the srsUE with the old (nr) and new (oran) gnb. 
Note that in the case of ORAN gnb, one might need to tune `time_adv_nsamples` in the `ue.conf.*` to get the best performance. 

Best,
Piotr
